.travis.yml

```diff
@@ -40,6 +40,8 @@ jobs:
     name: "E2E tests for CDC"
   - script: "./ci-scripts/lhci.sh"
     name: "Lighthouse score validation"
+  - script: "export CYPRESS_CONFIG=cypress.ci.epd-visualization.json && export SPA_ENV='b2b,epd-visualization' && ./ci-scripts/e2e-cypress.sh -s epd-visualization || exit 0"
```

Comment:
This creates  a Travis job that will run EPD e2es on each pull request for all Spartacus. Therefore, it should only be present in your integration branch. Please exclude it from the PR. More info [here](https://sap.github.io/spartacus-docs/contributing-integrations-to-spartacus/)

Reply:
What strategy is normally used to avoid a difference such as this from getting merged to the release/develop branch each time a PR to merge changes from the integration library branch to the release/develop branch?

Are you normally creating a branch off the integration library branch and then adding a commit to that to remove this change? Or cherry picking changes into a branch off the integration library branch?

Reply:
How about using a conditional job instead, so there is no need to maintain a difference between the branches?
I have found that this works nicely:

```yaml
dist: bionic
language: node_js
node_js:
  - 12.16.1
cache: yarn
git:
  depth: false
branches:
  only:
    - develop
    - "/^epic\\/.*$/"
    - "/^release\\/.*$/"
    - "/^integration-libs\\/.*$/"
defaults: &defaults
  script: "export CYPRESS_CONFIG=cypress.ci.json && export SPA_ENV='ci,b2c' && ./ci-scripts/e2e-cypress.sh"
jobs:
  include:
    - script: "./ci-scripts/unit-tests-core-lib.sh"
      name: "Unit tests for core and storefrontlib"
    - script: "./ci-scripts/unit-tests.sh"
      name: "Unit tests for libs"
    - script: "./ci-scripts/validate-lint.sh"
      name: "Validation checks"
    - script: "export CYPRESS_CONFIG=cypress.ci.b2b.json && export SPA_ENV='ci,b2b' && ./ci-scripts/e2e-cypress.sh -s b2b"
      name: "E2E regression tests for B2B"
    - script:
      name: "E2E tests for B2C, 1st job"
      <<: *defaults
    - script:
      name: "E2E tests for B2C, 2st job"
      <<: *defaults
    - script:
      name: "E2E tests for B2C, 3rd job"
      <<: *defaults
    - script:
      name: "E2E tests for B2C, 4th job"
      <<: *defaults
    - script: "./ci-scripts/e2e-cypress.sh --ssr"
      name: "E2E tests for SSR B2C"
    - script: "export CYPRESS_CONFIG=cypress.ci.cdc.json && export SPA_ENV='cdc' && ./ci-scripts/e2e-cypress.sh -s cdc || exit 0"
      name: "E2E tests for CDC"
    - script: "./ci-scripts/lhci.sh"
      name: "Lighthouse score validation"
    - script: "export CYPRESS_CONFIG=cypress.ci.epd-visualization.json && export SPA_ENV='b2b,epd-visualization' && ./ci-scripts/e2e-cypress.sh -s epd-visualization || exit 0"
      name: "E2E tests for EPD Visualization"
      if: branch IN (epic/epd-visualization, epic/epd-visualization-maintenance)
addons:
  chrome: stable
  sonarcloud:
    organization: sap
```

I can change to branches with an `integration-libs/` prefix instead of an `epic/` prefix in the branch name if that is preferred.

Reply:
How about using conditional jobs instead of maintaining a difference across branches?
I have found that this works nicely:

```yaml
dist: bionic
language: node_js
node_js:
  - 12.16.1
cache: yarn
git:
  depth: false
branches:
  only:
    - develop
    - "/^epic\\/.*$/"
    - "/^release\\/.*$/"
    - "/^integration-libs\\/.*$/"
defaults: &defaults
  script: "export CYPRESS_CONFIG=cypress.ci.json && export SPA_ENV='ci,b2c' && ./ci-scripts/e2e-cypress.sh"
jobs:
  include:
    - script: "./ci-scripts/unit-tests-core-lib.sh"
      name: "Unit tests for core and storefrontlib"
    - script: "./ci-scripts/unit-tests.sh"
      name: "Unit tests for libs"
    - script: "./ci-scripts/validate-lint.sh"
      name: "Validation checks"
    - script: "export CYPRESS_CONFIG=cypress.ci.b2b.json && export SPA_ENV='ci,b2b' && ./ci-scripts/e2e-cypress.sh -s b2b"
      name: "E2E regression tests for B2B"
    - script:
      name: "E2E tests for B2C, 1st job"
      <<: *defaults
    - script:
      name: "E2E tests for B2C, 2st job"
      <<: *defaults
    - script:
      name: "E2E tests for B2C, 3rd job"
      <<: *defaults
    - script:
      name: "E2E tests for B2C, 4th job"
      <<: *defaults
    - script: "./ci-scripts/e2e-cypress.sh --ssr"
      name: "E2E tests for SSR B2C"
    - script: "export CYPRESS_CONFIG=cypress.ci.cdc.json && export SPA_ENV='cdc' && ./ci-scripts/e2e-cypress.sh -s cdc || exit 0"
      name: "E2E tests for CDC"
    - script: "./ci-scripts/lhci.sh"
      name: "Lighthouse score validation"
    - script: "export CYPRESS_CONFIG=cypress.ci.epd-visualization.json && export SPA_ENV='b2b,epd-visualization' && ./ci-scripts/e2e-cypress.sh -s epd-visualization || exit 0"
      name: "E2E tests for EPD Visualization"
      if: branch IN (epic/epd-visualization, epic/epd-visualization-maintenance)
addons:
  chrome: stable
  sonarcloud:
    organization: sap
```

Also I can rename the branches to use a branch name prefix of `integration-libs/` instead of `epic/` if that is preferred.

Reply:
ah, that's a clean and nice approach. We could ask the other integration teams to just do the same.

Reply:
And yes, please use a branch prefix of `integration-libs`